dashboard/app/controllers/home_controller.rb

```diff
@@ -85,11 +85,32 @@ def init_homepage
         current_user.gallery_activities.order(id: :desc).page(params[:page]).per(GALLERY_PER_PAGE)
       @force_race_interstitial = params[:forceRaceInterstitial]
       @force_school_info_interstitial = params[:forceSchoolInfoInterstitial]
-      @recent_courses = current_user.recent_courses_and_scripts.slice(0, 2)
+      @recent_courses = current_user.recent_courses_and_scripts
+
+      unless current_user.teacher?
+        @recent_courses = current_user.recent_courses_and_scripts.drop(1)
+      end
```

Comment:
Because students have a separate component, StudentTopCourse, that handles all of the info about their most recent course. They only see CourseCards for courses/scripts beyond their most recent.  I'll add a comment explaining this.